shared/actions/teams.js

```diff
@@ -101,6 +101,32 @@ const _addPeopleToTeam = function*(action: TeamsGen.AddPeopleToTeamPayload) {
   yield Saga.put(createDecrementWaiting({key: Constants.teamWaitingKey(teamname)}))
 }
 
+const _getTeamRetentionPolicy = function*(action: TeamsGen.GetTeamRetentionPolicyPayload) {
+  const {teamname} = action.payload
+  yield Saga.put(createIncrementWaiting({key: Constants.teamWaitingKey(teamname)}))
+  const state: TypedState = yield Saga.select()
+  const teamID = Constants.getTeamID(state, teamname)
+  const policy: RPCChatTypes.RetentionPolicy = yield Saga.call(
+    RPCChatTypes.localGetTeamRetentionLocalRpcPromise,
+    {teamID}
+  )
+  let retentionPolicy: Types.RetentionPolicy = Constants.makeRetentionPolicy()
+  try {
+    retentionPolicy = Constants.serviceRetentionPolicyToRetentionPolicy(policy)
+    if (retentionPolicy.type === 'inherit') {
+      throw new Error(`RPC returned retention policy of type 'inherit' for team policy`)
+    }
+  } catch (err) {
+    logger.error(err.message)
+    throw err
+  } finally {
+    yield Saga.sequentially([
+      Saga.put(replaceEntity(['teams', 'teamNameToRetentionPolicy'], I.Map([[teamname, retentionPolicy]]))),
+      Saga.put(createDecrementWaiting({key: Constants.teamWaitingKey(teamname)})),
```

Comment:
none of these error flows decrement the waiting key

Reply:
why? the `finally` should run regardless

Reply:
oh sorry misread